src/Octoshift/GithubClient.cs

```diff
@@ -156,12 +162,24 @@ public virtual async IAsyncEnumerable<JToken> GetAllAsync(string url, Dictionary
             {
                 response.EnsureSuccessStatusCode();
             }
-            else if (response.StatusCode != status)
+            else if (response.StatusCode != status && RateLimitRemaining(headers) != 0)
             {
                 throw new HttpRequestException($"Expected status code {status} but got {response.StatusCode}", null, response.StatusCode);
             }
 
-            return (content, response.Headers.ToArray());
+            CheckForRetryDelay(headers);
+
+            return (content, headers);
+        }
+
+        private async Task ApplyRetryDelayAsync()
+        {
+            if (_retryDelay > 0.0)
```

Comment:
two nits:
1. there is no need to necessarily compare two `doubles` in C# the case is going to happen automatically otherwise  you're gonna get a compiler error clearly stating that there's no implicit case available for the comparison.
2. For literals it's better (more clear) to use their respective suffix, for more info take a look at [this](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/integral-numeric-types). So for double (which is the default for floating points) I'd write `0D` instead of `0.0`. Of course if you use explicit typing there is no need to use the suffix (eg. `double dbl = 1.5`)